### ISSUE TITLE
Request Query params breaking when getting Collection.toJSON()

### ISSUE THREAD

**Issue by melloflavio** [postmanlabs/postman-collection#254]
I'm trying to use postman-collection to programatically generate postman collections. However I've been experiencing some errors.

For example, the following code:

```javascript
    var collection = new Postman.Collection();
    var item = new Postman.Item({"id":"52a377ae-d3af-4000-8ed0-ba71c21912e1","response":[],"event":[],"name":"GET - /integers/","request":{"url":"https://www.random.org:443/integers/?num=1&min=0&max=255&col=16&base=10&format=plain&rnd=new","method":"GET"}})
    collection.items.add(item);
   fs.writeFileSync("myCollection.postman_collection", JSON.stringify(collection.toJSON()));
```

produces the following postman collection:

```json
{
    "event": [],
    "info": {
        "id": "310cb9eb-14c2-47a5-b3e2-7ca1caa73818",
        "schema": "https://schema.getpostman.com/json/collection/v2.0.0/collection.json"
    },
    "item": [
        {
            "_": {
                "postman_requiresId": true
            },
            "event": [],
            "id": "52a377ae-d3af-4000-8ed0-ba71c21912e1",
            "name": "GET - /integers/",
            "request": {
                "method": "GET",
                "url": "https://www.random.org:443/integers/?members=num=1,min=0,max=255,col=16,base=10,format=plain,rnd=new&reference=[object Object]&Type=function PostmanQueryParam(options) {\n        // this constructor is intended to inherit and as such the super constructor is required to be excuted\n        QueryParam.super_.apply(this, arguments);\n\n        this.update(options);\n    }"
            },
            "response": []
        }
    ],
    "variable": []
}
```

notice how the query parameters get completely mangled in the process. They are separated by `,` instead of `&`. Also, there is a whole new section being added in the query string, which seems to be part of the source code: 
```
(...) &reference=[object Object]&Type=function PostmanQueryParam(options) {\n        // this constructor is intended to inherit and as such the super constructor is required to be excuted\n        QueryParam.super_.apply(this, arguments);\n\n        this.update(options);\n    }
```

**Comment by czardoz** [comment]
Thanks for reporting this. This is an issue with the automatic typecasting when you are doing `collection.items.add(item);`. We'll work on a fix for the next release. 

In the meantime, you can do the following workaround: Instead of passing an instance of `Postman.Item` to `collection.items.add`, you can pass the plain JS object as is:

```
collection.items.add({
    "id": "52a377ae-d3af-4000-8ed0-ba71c21912e1",
    "response": [],
    "event": [],
    "name": "GET - /integers/",
    "request": {
        "url": "https://www.random.org:443/integers/?num=1&min=0&max=255&col=16&base=10&format=plain&rnd=new",
        "method": "GET"
    }
});
```

**Comment by kunagpal** [comment]
@melloflavio A fix for this bug has been added in #234, we'll update this thread when it has been released.

**Comment by kunagpal** [comment]
@melloflavio A fix for this bug was released with `postman-collection@1.2.5`. 😄
